app_spec/zomes/blog/code/src/post.rs

```diff
@@ -53,9 +56,22 @@ pub fn definition() -> ValidatingEntryType {
             hdk::ValidationPackageDefinition::ChainFull
         },
 
-        validation: |post: crate::post::Post, _ctx: hdk::ValidationData| {
-            (post.content.len() < 280)
-                .ok_or_else(|| String::from("Content too long"))
+        validation: |data: ValidationData<Post>| {
+            match data {
+                ValidationData::Create { entry, .. } => (entry.content.len() > 280)
+                    .ok_or_else(|| "Content too long".into()),
+                ValidationData::Modify { .. } => Ok(()),
+                ValidationData::Delete { entry_addr, orig_entry, .. } => {
+                    let del_entry = hdk::get_entry(&entry_addr)?;
+                    let orig_entry = hdk::get_entry(&orig_entry)?;
+
+                    let del_entry_sources = entry.sources();
+                    let orig_entry_sources = entry.sources();
+
+                    (del_entry_sources == orig_entry_sources)
+                        .ok_or_else(|| "deletion is unauthorized".into())
```

Comment:
Would be cool to have a test that triggers this and verifies entries can only be deleted by their authors..